### DISCUSSION TITLE
@urql/vue -- How to reset cache?

### DISCUSSION THREAD

**Discussion by bartenra** [urql-graphql/urql#1961] [Q&A] [answered]
On logout, we need to reset the entire cache so that the next user cannot access the data of the user that was just logged in (and sees their own data instead).

There is [this thread](https://github.com/FormidableLabs/urql/issues/297) with an approach that works for React. There's also my reply with an approach which I thought would be applicable for Vue, which turns out to be wrong -- `provideClient` can only be called inside `setup`, not in a `watch`. So creating a new client is not an option for Vue, since the new client cannot be `provide`d.

We're now moving towards production and found this out. So I'm wondering:

1. Is there another way to reset the entire cache? If not, should there be?
2. Alternatively, should it be possible to pass a `Ref<Client>` to `provideClient`?

**Comment by bartenra** [comment]
Our workaround now is to just do a page refresh on logout.

**Comment by kitten** [comment, marked as answer]
> Alternatively, should it be possible to pass a Ref<Client> to provideClient?

This sounds like an oversight on my end. That should definitely be possible, i.e. added to the implementation

**Reply by bartenra** [reply]
I see it's been added! :tada:  https://github.com/FormidableLabs/urql/releases/tag/%40urql%2Fvue%400.5.0

**Comment by moeriki** [comment]
@bartenra did you make it work without a refresh?

@kitten looking at the code I'm puzzled how this should work.

```ts
  let client = opts.value;
  app.provide('$urql', client);
```

If we take the value out of the ref before providing it, the provided client won't be updated if the ref value changes right?

https://v3.vuejs.org/guide/composition-api-provide-inject.html#reactivity

> To add reactivity between provided and injected values, we can use a ref or reactive when providing a value.

I think the ref should be provided as-is. This would certainly have implications on the whole API.
